### PR TITLE
Add configuration for hypervisor block storage driver

### PR DESCRIPTION
**Pull request by amshinde**

### CODE REVIEW

**Comment by sboeuf** [comment]
LGTM

File: config/configuration.toml.in [hunk 1]
```diff
@@ -64,6 +64,11 @@ default_bridges = @DEFBRIDGES@
 # 9pfs is used instead to pass the rootfs.
 disable_block_device_use = @DEFDISABLEBLOCK@
 
+# Block storage driver to be used for the hypervisor in case the container
+# rootfs is backed by a block device. This is either virtio-scsi or 
+# virtio-blk.
+block_driver_driver = @DEFBLOCKSTORAGEDRIVER@
```
Comment (jodh-intel) [inline config/configuration.toml.in]: This needs to be double-quoted, as currently:

```
$ ./cc-runtime cc-env                                                                                                                                                   
Near line 70 (last key parsed 'hypervisor.qemu.block_driver_driver'): expected value but found "virtio" instead
```

File: cc-env_test.go [hunk 2]
```diff
@@ -68,6 +68,7 @@ func makeRuntimeConfig(prefixDir string) (configFile string, config oci.RuntimeC
 	shimPath := filepath.Join(prefixDir, "cc-shim")
 	proxyPath := filepath.Join(prefixDir, "cc-proxy")
 	disableBlock := true
+	blockStorageDriver := "virtio-scsi"
```
Comment (jodh-intel) [inline cc-env_test.go]: Seeing this change makes me wonder if we should be displaying it in the ` [Hypervisor]` section of the `cc-runtime cc-env` output?

File: config.go [hunk 3]
```diff
@@ -218,6 +219,14 @@ func (h hypervisor) defaultBridges() uint32 {
 	return h.DefaultBridges
 }
 
+func (h hypervisor) blockDeviceDriver() string {
+	if h.BlockDeviceDriver == "" {
+		return defaultBlockDeviceDriver
+	}
+
+	return h.BlockDeviceDriver
```
Comment (devimc) [inline config.go]: You should validate the block device  `virtio-scsi or virtio-blk`

File: cc-env_test.go (hunk 2 above)
Comment (amshinde) [reply cc-env_test.go]: Makes sense to show this in the cc-env output. I have included this in the PR now.

File: config.go (hunk 3 above)
Comment (amshinde) [reply config.go]: Done

**Comment by amshinde** [comment]
@chavafg I see kubernetes failing here. Is this a valid faliure? Can you post the logs?

**Comment by devimc** [comment]
check-go-static failed

```
 package honnef.co/go/tools/cmd/unused: unrecognized import path "honnef.co/go/tools/cmd/unused" (parse https://honnef.co/go/tools/cmd/unused?go-get=1: no go-import meta tags 
```

**Comment by chavafg** [comment]
@amshinde k8s failure was a network timeout, so not valid. BTW, we just switch to run k8s validation through the jenkins jobs, so no more qa-fails from clearcontainersbot

File: cc-env_test.go (hunk 2 above)
Comment (jodh-intel) [reply cc-env_test.go]: Thanks! Could you bump `formatVersion` to `1.0.9` as this is a `cc-env` format change.

**Comment by amshinde** [comment]
@jodh-intel Can you approve and merge this PR if this looks good to you.

**Comment by jodh-intel** [comment]
Thanks @amshinde.

lgtm.

**Pull request merged**